adr: boot disk size (#504)
elastisys/compliantkubernetes: docs/adr/0032-boot-disk-size.md

# Boot disk size on nodes

* Status: accepted
* Deciders: Cristian Klein, Olle Larsson, Lucian Vlad, Fredrik Liv, Robin Wallace, Pavan Gunda
* Date: 2023-02-14

## Context and Problem Statement

Our current setup is using boot disks of 50GB and we noticed that for nodes of bigger sizes that is not always enough. We also noticed that the space is filled up quickly by the contaner images.

> We woul like to have same boot disk sizes for all our nodes on all the cloud providers if possible.

Reviewer: ```suggestion
We would like to have same boot disk sizes for all our nodes on all the cloud providers if possible.
```

Author: Fixed
Thx

Should we increase the boot disk size to a bigger size?

## Decision Drivers

* We want to best serve our customer needs.

> * We want to find a solution which is scalable and minimizes administrator burden.

Reviewer: Maybe also:

```
* We don't want to waste infrastructure.
```

# Boot disk size on nodes

* Status: accepted
* Deciders: Cristian Klein, Olle Larsson, Lucian Vlad, Fredrik Liv, Robin Wallace, Pavan Gunda
* Date: 2023-02-14

## Context and Problem Statement

> Our current setup is using boot disks of 50GB and we noticed that for nodes of bigger sizes that is not always enough. We also noticed that the space is filled up quickly by the contaner images.

Reviewer 2: ```suggestion
We have often defaulted to using boot disks of 50GB where possible but as of late we have noticed that for some environments this is not sufficient. We also noticed that the available space is commonly filled up by customer container images.
```

Author: Fixed.
Please recheck
Thanks

## Decision Drivers

* We want to best serve our customer needs.
* We want to find a solution which is scalable and minimizes administrator burden.
* We don't want to waste infrastructure.

## Considered Options

* Keep boot disk size to 50Gb but increase the alert threshold and do regular cleanup
* Keep boot disk size to 50Gb and increase boot disk size only where is needed
* Increase the boot disk size to 100GB for all nodes irrespective of node size
* Increase the boot disk size to 100GB only for nodes that are bigger than 4C8GB
* Use local disk of 100GB size for controlplane nodes

## Decision Outcome

Chosen option: Increase the boot disk size to 100GB for all nodes irrespective of node size & Use local disk of 100GB size for controlplane nodes

### Positive Consequences

* There will be more available space for pulling images
* Reduce the number of alerts and ops time
* Improve platform stability and scalability

### Negative Consequences

* We need to replace all existing nodes for our existing environments
* Cost will increase depending on number of nodes and price per GB of storage

## Recommendation to Operators

* Try to use same VM flavors on all environments

> * Use VM flavor with local disk of 100GB or whichever is closest to this size depending on cloud provider.

Reviewer 3: Didn't we say that we should try to use "boot from volume" rather than local disk? except for control-plane nodes

```suggestion
* Use VM flavor with local disk of 100GB or whichever is closest to this size depending on cloud provider for controlplane nodes.
```

Author: fixed
